### PR TITLE
Inject the identifier to be used into WL.

### PR DESCRIPTION
**Pull request by nithyaganeshng**
**Problem:**
Currently, WL creates an identifier and uses it by default for signing the payload. 

**Solution:**

- Support to inject the identifier to be used by WL.
- Make the existing identifier as backup option in case no identifier is injected.
- Use the first identifier from the list of injected identifiers to sign the payload.
- Fix the failing unit tests.
- Clean up code by removing unused code. 

**Validation:**
Please include here how it was verified that the PR works as intended.

**Type of change:**
- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/networking/formatters/OpenId4VciIssuanceRequestFormatter.kt [hunk 1]
```diff
@@ -53,15 +55,17 @@ internal class OpenId4VciIssuanceRequestFormatter(private val libraryConfigurati
...
-            serializedResponseContent,
-            responder,
-            com.microsoft.walletlibrary.util.Constants.OPENID4VCI_TYPE_HEADER
-        )
+        formatJwsHeaderForIdentifier(holderIdentifier)
+        return holderIdentifier.sign(serializedResponseContent)
+    }
+
+    private fun formatJwsHeaderForIdentifier(holderIdentifier: HolderIdentifier) {
+        val jwsHeader = JwsHeaderFormatter.formatHeader(holderIdentifier, com.microsoft.walletlibrary.util.Constants.OPENID4VCI_TYPE_HEADER)
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/walletlibrary/networking/formatters/OpenId4VciIssuanceRequestFormatter.kt:62]: I would want a double-check on this but isn't this type value here the `typ` header, which we've always had `JWT`?

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/serializer/PresentationExchangeResponseBuilder.kt [hunk 2]
```diff
@@ -38,78 +38,70 @@ internal class PresentationExchangeResponseBuilder(
...
+            audience = clientId
+            nonce = requestNonce
+            responseCreationTime = issuedTime
+            responseExpirationTime = expiryTime
         }
-        // unreachable
-        return ""
+
+        val token = libraryConfiguration.serializer.encodeToString(PresentationResponseClaims.serializer(), oidcResponseClaims)
+        return identifier.sign(token)
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/serializer/PresentationExchangeResponseBuilder.kt]: I don't think this returns the expected value. Comparing with iOS, I think we may be returning different results for the HolderIdentifier's `sign` function. It seems we're returning a complete JWS token here, but iOS returns just the signature.

File: walletlibrary/src/test/java/com/microsoft/walletlibrary/did/sdk/RevocationServiceTest.kt [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (logangirvin) [inline walletlibrary/src/test/java/com/microsoft/walletlibrary/did/sdk/RevocationServiceTest.kt]: Can we just delete this and the IdentifierManager files that are completely commented out?

**Review by logangirvin — CHANGES_REQUESTED** [review]
I think we might have a substantial difference with the mean intended return value of IdentityHolder's sign function that should be resolved, which unfortunately means a few layers of changes between different request protocols.

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/serializer/PresentationExchangeResponseBuilder.kt (hunk 2 above)
Comment (nithyaganeshng) [reply walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/serializer/PresentationExchangeResponseBuilder.kt]: Changed the implementation to return only the signature.

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/networking/formatters/OpenId4VciIssuanceRequestFormatter.kt (hunk 1 above)
Comment (nithyaganeshng) [reply walletlibrary/src/main/java/com/microsoft/walletlibrary/networking/formatters/OpenId4VciIssuanceRequestFormatter.kt:62]: We use a different typ header for OIDC4VCI protocol. We use "JWT" for other flows.

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/identifier/HolderIdentifier.kt [hunk 4]
```diff
@@ -25,5 +25,5 @@ interface HolderIdentifier {
      * @param data The data to sign.
      * @return The signed data.
      */
-    fun sign(data: String): String
+    fun sign(data: ByteArray?): String
```
Comment (symorton) [inline walletlibrary/src/main/java/com/microsoft/walletlibrary/identifier/HolderIdentifier.kt]: Can this return a ByteArray to be in parity with iOS? and then an injected Identifier doesnt have to worry about converting the signature into the properly encoded string? 

Also, why is the input nullable?

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/identifier/EncryptedSharedPreferencesIdentifier.kt [hunk 5]
```diff
@@ -13,13 +14,14 @@ internal class EncryptedSharedPreferencesIdentifier(
...
     override val keyReference: String,
-    private val cryptoOperations: CryptoOperations,
     private val keyStore: EncryptedKeyStore
 ) : HolderIdentifier {
 
-    override fun sign(data: String): String {
-        val keyId = "$id#$keyReference"
-        val privateKey = keyStore.getKey(keyReference)
-        return cryptoOperations.sign(data, privateKey, algorithm, keyId)
+    var jwsHeader: JWSHeader? = null
```
Comment (symorton) [inline walletlibrary/src/main/java/com/microsoft/walletlibrary/identifier/EncryptedSharedPreferencesIdentifier.kt]: Is the jwsHeader always null? What if we use the algorithm reference in the Identifier to construct the proper JwsHeader needed?

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/crypto/protocols/jose/jws/ECDSASignerWrapper.kt [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    override fun getJCAContext(): JCAContext {
+        TODO("Not yet implemented")
+    }
+
+    override fun supportedJWSAlgorithms(): MutableSet<JWSAlgorithm> {
+        return mutableSetOf(JWSAlgorithm(holderIdentifier.algorithm))
+    }
+
+    override fun sign(header: JWSHeader?, signingInput: ByteArray?): Base64URL {
+        (holderIdentifier as EncryptedSharedPreferencesIdentifier).jwsHeader = header
```
Comment (symorton) [inline walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/crypto/protocols/jose/jws/ECDSASignerWrapper.kt]: is this temporary? What happens when a dev tries to inject a different type of Identifier into the librarly?

File: walletlibrary/src/test/java/com/microsoft/walletlibrary/did/sdk/identifier/resolvers/MockRootOfTrustResolverFailure.kt [hunk 7]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved
+
+package com.microsoft.walletlibrary.did.sdk.identifier.resolvers
+
+import com.microsoft.walletlibrary.did.sdk.identifier.models.identifierdocument.DidMetadata
```
Comment (symorton) [inline walletlibrary/src/test/java/com/microsoft/walletlibrary/did/sdk/identifier/resolvers/MockRootOfTrustResolverFailure.kt]: why create three different objects for this? Can you pass the expected result in the constructor?

**Review by symorton — CHANGES_REQUESTED** [review]

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/identifier/EncryptedSharedPreferencesIdentifier.kt (hunk 5 above)
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/walletlibrary/identifier/EncryptedSharedPreferencesIdentifier.kt]: Why is this parameter necessary? the header is passed to the ecdsaSigner, but poking into the signer it just wants to know what algorithm to use, so couldn't we construct it out of our alg parameter?

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/crypto/protocols/jose/jws/ECDSASignerWrapper.kt (hunk 6 above)
Comment (logangirvin) [reply walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/crypto/protocols/jose/jws/ECDSASignerWrapper.kt]: I think this can be removed.

**Review by logangirvin — DISMISSED** [review]

**Review by symorton — APPROVED** [review]

**Review by logangirvin — APPROVED** [review]

**Pull request merged**